Header header 

uint64 loopKeyCur
uint64[] loopKeyPre